sources/packages/backend/apps/api/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -103,4 +103,49 @@ export class CASSupplierService {
 
     return student.casSupplier;
   }
+
+  /**
+   * Retries CAS supplier info for a student.
+   * Inserts a new CAS Supplier record for the student with PendingSupplierVerification status.
+   * @param studentId student id.
+   * @param auditUserId user id for the record creation.
+   */
+  async retryCASSupplier(
+    studentId: number,
+    auditUserId: number,
+  ): Promise<CASSupplier> {
+    const student = await this.studentService.getStudentById(studentId);
+    const auditUser = { id: auditUserId } as User;
+    if (!student) {
+      throw new CustomNamedError("Student not found.", STUDENT_NOT_FOUND);
+    }
+    const studentCasSupplier: CASSupplier = await this.casSupplierRepo.findOne({
+      select: {
+        id: true,
+        createdAt: true,
+        supplierStatus: true,
+      },
+      where: {
+        student: { id: studentId },
+      },
+      order: {
+        createdAt: "DESC",
+      },
+    });
+    if (
+      studentCasSupplier?.supplierStatus ===
+      SupplierStatus.PendingSupplierVerification
+    ) {
+      throw new UnprocessableEntityException(
+        `There is already a CAS Supplier for this student in ${SupplierStatus.PendingSupplierVerification} status.`,
+      );
```

Comment:
Only API controller and API controller services throw HTTP errors.
It should throw a custom error and be converted at the controller to the appropriate HTTP error.